ext/semian/tickets.c

```diff
@@ -16,8 +55,15 @@ update_ticket_count(update_ticket_count_t *tc)
   if (get_sem_val(tc->sem_id, SI_SEM_CONFIGURED_TICKETS) != tc->tickets) {
```

Comment:
You already have the lock, so I don't think you need the second check. I think this is a relic from the time when the logic was: `check if wrong -> grab lock -> check if still wrong -> adjust if still wrong`. Now we just always grab the lock and only do adjustments at register-time, which is simpler.